coordinator/src/models/errors/validation-errors.ts

```diff
@@ -3,77 +3,33 @@ import {MessageType} from "../../routes/util"
 export interface ValidationError {
   message: string
   operationOutcomeCode: "value"
-  apiErrorCode: string
   severity: "error" | "fatal"
+  expression: Array<string>
 }
 
-class ValueError implements ValidationError {
+export class MedicationRequestValueError implements ValidationError {
   message: string
   operationOutcomeCode = "value" as const
-  apiErrorCode = "MISSING_FIELD"
   severity = "error" as const
-}
-
-export class ContainsExactlyError extends ValueError {
-  constructor(number: number, resourceType: string) {
-    super()
-    this.message = `Bundle must contain exactly ${number} resource(s) of type ${resourceType}.`
-  }
-}
-
-export class ContainsBetweenError extends ValueError {
-  constructor(min: number, max: number, resourceType: string) {
-    super()
-    this.message = `Bundle must contain between ${min} and ${max} resource(s) of type ${resourceType}.`
-  }
-}
-
-export class ContainsAtLeastError extends ValueError {
-  constructor(number: number, resourceType: string) {
-    super()
-    this.message = `Bundle must contain at least ${number} resource(s) of type ${resourceType}.`
-  }
-}
-
-export class MissingIdError extends ValueError {
-  message = "ResourceType Bundle must contain 'id' field."
-}
-
-export class MedicationRequestValueError extends ValueError {
-  apiErrorCode = "INVALID_VALUE"
+  expression: Array<string>
 
   constructor(fieldName: string, uniqueFieldValues: Array<string>) {
-    super()
     this.message = `Expected all MedicationRequests to have the same value for ${
       fieldName
     }. Received ${[
       ...uniqueFieldValues
     ]}.`
+    this.expression = [`Bundle.entry.resource.ofType(MedicationRequests).${fieldName.split(" ")[0]}`]
```

Comment:
Might be a good idea to use the field name as passed in. Then the field name could be something like:
```
Bundle.entry.resource.ofType(MedicationRequest).extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")
```

Reply:
Or at least change the field names to valid FHIR expressions, e.g.:
```
extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")
```

Reply:
(I know I wrote the field names in the first place :shushing_face:)